resources/scripts/non_spacing_marks.py

```diff
@@ -0,0 +1,58 @@
+"""glyphsLib uses GlyphData.xml to know things about glyphs.
+
+Naively you would think we could get the same information by querying unicode
+properties but it seems not, for example brevecomb-cy with no mapped codepoint
+can be identified as a non-spacing mark by name. No unicode db will do that for us.
+
+This utility generates a Rust file with key information extracted from GlyphData.xml.
+"""
+
+from absl import app
+from lxml import etree
+from pathlib import Path
+import sys
+
+
+def main(_):
+    glyphs_lib = Path("../glyphsLib")
+    if not glyphs_lib.is_dir():
+        sys.exit(f"Missing {glyphs_lib}")
+    glyphdata_file = glyphs_lib / "Lib/glyphsLib/data/GlyphData.xml"
+    if not glyphdata_file.is_file():
+        sys.exit(f"Missing {glyphdata_file}")
+    
+    codepoints = set()
+    names = set()
+    xml = etree.parse(glyphdata_file)
+    for glyph in xml.xpath("//glyph[@category='Mark' and @subCategory='Nonspacing']"):
+        unicode = glyph.attrib.get("unicode")
+        if unicode is not None:
+            codepoints.add(unicode)
+        name = glyph.attrib.get("name")
+        if name:
+            names.add(name)
+
+    with open("glyphs2fontir/src/glyphdata.rs", "w") as f:
+        f.write("//! Generated module with Glyphs-specific data\n")
+        f.write("//!\n")
+        f.write("//! Run resources/scripts/non_spacing_marks.py to update\n")
+        
+        f.write("\n");
+        f.write("use std::collections::HashSet;\n")
+        f.write("\n")
+        f.write("pub(crate) fn is_nonspacing_mark(codepoints: &HashSet<u32>, name: &str) -> bool {\n")
+        f.write("    match codepoints.iter().next() {\n")
```

Comment:
wouldn't it be preferable to use HashSets instead of a match here? you already know for which codepoints or glyph names GlyphData.xml sets those two attributes, you could hard-code them in this file (maybe using https://crates.io/crates/lazy_static) and then just check if any of the inputs is in those sets in O(1) time, I believe.

Reply:
notes that glyphsLib (and probably Glyphs.app too) looks things up by glyph name first and only when that's not found it looks them up by unicode so you should invert the ordering of the checks you have here.

Also I see with `codepoints.iter().next()` you are basically only checking one codepoint if any, the first one, and it is even a HashSet so I believe the order in which you iterate is random? glyphsLib looks up all the codepoints, not just the first, and breaks at the first found. I believe the `unicodes` is an ordered list rather than a set in in Glyphs (like in UFO, even if it doesn't make any sense because a "primary" codepoint is meaningless, cmap maps from codepoint to glyphs not viceversa).

Anyway, look up by name and if that's not found try to look up by unicode, all of them not just the random first -- this should match glyphsLib, at least for the data actually recorded in the XML files. There's other stuff that is made up at runtime when look up fails (see `_construct_production_name` and `_construct_category` in `glyphsLib.glyphdata`), but we can tackle that later)

Reply:
> wouldn't it be preferable to use HashSets instead of a match here

I would expect the compiler output for the match to be _at least_ as good, probably better, than a general purpose hashset. We're giving it more to work with, it knows all the possible items. Also HashSet doesn't support const construction so we want something like [oncecell](https://doc.rust-lang.org/std/cell/struct.OnceCell.html) but that's still nightly/experimental.

> glyphsLib looks up all the codepoints, not just the first, and breaks at the first found

That's a bug, will fix. I assumed unicodes to always contain *one* codepoint for no particularly good reason.

Reply:
I don't have the original ordering handy, if we need that I'd prefer to tackle it in a separate issue.

Reply:
it should be fine